cloudpathlib/s3/s3client.py

```diff
@@ -199,10 +199,11 @@ def _list_dir(self, cloud_path: S3Path, recursive=False) -> Iterable[Tuple[S3Pat
 
                 # files in the directory
                 for result_key in result.get("Contents", []):
-                    yield (
-                        self.CloudPath(f"s3://{cloud_path.bucket}/{result_key.get('Key')}"),
-                        False,
-                    )
+                    if result_key.get("Size") > 0:
+                        yield (
+                            self.CloudPath(f"s3://{cloud_path.bucket}/{result_key.get('Key')}"),
+                            False
+                        )
```

Comment:
So I think what we want here is a

```python
if result_key.get("Key").endswith("/") and result_key.get("Size") == 0:
    yield (..., True)
else:
    yield (..., False)
```

This way, we cover both empty files and fake directories. 

However, I think we may want to use the `yielded_dirs` as with the non-recursive case, to avoid returning duplicates if the fake directory came up from having children.